integrations/tensorflow/bindings/python/pyiree/tf/support/tf_utils.py

```diff
@@ -269,32 +280,40 @@ def __init__(self,
     super().__init__(module_class, backend_info, exported_names, artifacts_dir)
 
     set_random_seed()
-    self._module_blob, self.compiled_path = compile_tf_module(
+    self._module_blob, compiled_path = compile_tf_module(
         tf_module=module_class(),
         backend_infos=[backend_info],
         exported_names=exported_names,
         artifacts_dir=artifacts_dir)
     self._module = rt.VmModule.from_flatbuffer(self._module_blob)
     self._config = rt.Config(driver_name=backend_info.driver)
 
+    self.compiled_paths = None
+    if compiled_path is not None:
+      if not len(exported_names):
+        # Get all method names on 'module_class' that aren't on 'tf.Module'.
+        exported_names = _get_non_inhereted_function_names(module_class)
```

Comment:
methods on subobjects can also be exported We should document this restriction for users and leave a TODO, or change _get_non_inhereted_function_names to walk child objects (I think that tf.Module or one of its serialization helpers might already have this introspection capability since that is what it needs to do anyway when saving to SavedModel?). btw, a more robust thing to do here might be to see which properties are @tf.function's, tf.Variable's and tf.Constant's, rather than just what properties are not in parents. That's the true logic which matches the SavedModel semantics.

btw, more complicated situations can happen too: 
Here's an example where the same method is reachable by two exported names:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/mlir/tensorflow/tests/tf_saved_model/dag_object_graph.py

I'm fine with this for now though. We just need to document the restriction, as this is a correctness issue. If possible, aborting the program in cases where we aren't sure would be better (see if any of the properties are tf.Module's themselves and conservatively abort; that's probably just as hard to write as a simple recursive traversal of tf.Module's though).

Reply:
Yeah, I definitely wrote this as a quick way to pretend like TFLite can compile entire `tf.Module`s (which it doesn't even via `SavedModel`s). I'm okay with not supporting the more nuanced functionality for `TfLiteCompiledModule`, since TFLite itself doesn't really support that from the perspective of end-to-end compilation of `tf.Module`s.

I'm not a fan of adding a restriction to `IreeCompiledModule` just add a dictionary what might as well be `"any string" --> compiled.vmfb path` though. I'm looking into if I can get a list of the actual functions that the `IreeCompiledModule` has access to to use here instead. For now though I would like to get this change through so that I can get some better documentation out. I'll put a TODO into the IREE module for the time being.